planner/core/logical_plan_builder.go

```diff
@@ -157,7 +157,6 @@ func (b *PlanBuilder) buildResultSetNode(node ast.ResultSetNode) (p LogicalPlan,
 			col.OrigTblName = col.TblName
 			if x.AsName.L != "" {
 				col.TblName = x.AsName
-				col.DBName = model.NewCIStr("")
```

Comment:
This line is the real change in this PR

Reply:
why add an empty database name can fix the issue?

Reply:
> One of the visitInfo record is
> 
> ```
> db=""
> table="t1"
> user="test"
> ```
> 
> It doesn't match any privilege record in the cache, so privilege check fail.
> 
> 

@zz-jason

Reply:
db name is not set correctly...

Reply:
The db name is set to "", and we should not set it to ""